evalml/tests/automl_tests/test_automl_iterative_algorithm.py

```diff
@@ -0,0 +1,1146 @@
+import json
+import warnings
+from unittest.mock import MagicMock, patch
+
+import numpy as np
+import pandas as pd
+import pytest
+from skopt.space import Categorical, Integer
+
+from evalml import AutoMLSearch
+from evalml.automl.automl_algorithm import IterativeAlgorithm
+from evalml.automl.automl_algorithm.iterative_algorithm import (
+    _ESTIMATOR_FAMILY_ORDER,
+)
+from evalml.automl.callbacks import raise_error_callback
+from evalml.exceptions import ParameterNotUsedWarning
+from evalml.model_family import ModelFamily
+from evalml.pipelines import (
+    BinaryClassificationPipeline,
+    ComponentGraph,
+    Estimator,
+)
+from evalml.pipelines.components.utils import (
+    allowed_model_families,
+    get_estimators,
+)
+from evalml.problem_types import ProblemTypes
+from evalml.tuners import SKOptTuner
+
+
+def test_automl_feature_selection_with_allowed_component_graphs_iterative(
```

Comment:
In general anything that uses `max_iterations`, `ensembling`, `allowed_component_graphs` or `allowed_model_families` was tagged as `_iterative` and moved to these files as they are only used by `IterativeAlgorithm`.